DietPi-Update | Switch applying pre-patch and new versioning system between patch_file and pre-patch_file

MichaIng opened the pull request:
This is just an alternative. Perhaps looks a bid cleaner and allows applying pre-patches before applying new versioning system. Either way is okay for me, the way it is currently in dev and the way this PR does.

**Status**: Ready

**Commit list/description**:
+ DietPi-Pre-patch | Minor coding, visual and exit code info
+ DietPi-Patch | Minor coding and wording

MichaIng committed:
v6.17

+ DietPi-Pre-patch | Apply new versioning and branch system within pre-patch_file

MichaIng committed:
v6.17

+ DietPi-Patch | Apply new versioning and branch system within pre-patch_file, run pre-patch instead from patch_file

MichaIng committed:
v6.17

+ DietPi-Pre_patch | Minor visual

MichaIng committed:
v6.17

+ DietPi-Pre_patch | Minor

Fourdee committed:
Fixes

Syntax and init blank value.

Fourdee commented:
@MichaIng 

I believe we should keep the version system update patch in ```patch_file```:
- ```patch_file``` code is reliable for pre RC version system. And, it forces the user to re-run ```dietpi-update``` which has the new ```pre-patch_file``` code in, and enables the new versioning system at the same time.
- If we make major changes to ```pre-patch_file``` in the future, we could break this update to new version system. Whereas, the core method of ```patch_file``` system will likely never be changed.

MichaIng commented:
@Fourdee 
Yeah, as said, both would work for my impression. With this PR, other pre-patch issues can be fixed a bid earlier, but since the update already needed to succeed until `patch_file` execution, it should also succeed until regular `pre-patch_file` execution of 2. manual `dietpi-update` run.

But yeah with this PR it is one level more nested. I will send commit to revert, but keep the other minor changes.

€: Okay done. The more I think about it the more I think this solution is indeed the better/safer one.
Btw. not sure if it makes sense, but we could move pre-patch_file to another location, so it is not by default copied to /DietPi/dietpi/.

MichaIng committed:
v6.17

+ DietPi-Patch | Revert applying new version system inside patch_file

MichaIng committed:
v6.17

+ DietPi-Patch | Minor alignment

MichaIng committed:
v6.17

+ DietPi-Pre-patch | Revert to apply new version system in patch_file again
+ DietPi-Pre-patch | Add info about exit code representing ID of failed pre-patch, in case

MichaIng committed:
v6.17

+ DietPi-Update | Add exit code (failed pre-patch ID) info to pre-patch error message

MichaIng committed:
v6.17

+ DietPi-Patch | Remove "pre-patch_file", which will be moved to RAMdisk by pre-v6.17 dietpi-update.

MichaIng committed:
v6.17

+ DietPi-Patch | Minor

Fourdee commented:
@MichaIng 

Thanks for the changes 👍 

> Btw. not sure if it makes sense, but we could move pre-patch_file to another location, so it is not by default copied to /DietPi/dietpi/.

We could do, but I believe its fine where it is, next to ```patch_file```, and, its deleted afterwards on local system so no harm.
